### PR TITLE
feat(make:factory): use autocompletion for no persistence classes

### PR DESCRIPTION
**Pull request by nikophil**
related to #336 

When no `--no-persistence` is passed we ask for a FQCN, which could be cumbersome to write.

Then this PRs proposes as autocomplete all classes found in compoer's PSR-4 config

Peek 14-12-2022 09-01

### CODE REVIEW

File: src/Bundle/Maker/Factory/NoPersistanceObjectsAutoCompleter.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+{
+    public function __construct(private string $kernelRootDir)
+    {
+    }
+
+    public function getAutocompleteValues(): array
+    {
+        $classes = [];
+
+        $excludedFiles = $this->excludedFiles();
```
Comment (nikophil) [inline src/Bundle/Maker/Factory/NoPersistanceObjectsAutoCompleter.php:17]: the file `src/functions.php` was problematic: when transforming this file into its PSR4 representation, if I call `class_exists()`, I got an error I can't even catch as a `\Throwable` :shrug: 

```
Fatal error: Cannot redeclare Zenstruck\Foundry\factory() (previously declared in /app/src/functions.php:16) in /app/src/functions.php on line 16
```

then I decided to exclude all files found in composer's config under `autoload.files` node

File: tests/Unit/Bundle/Maker/Factory/NoPersistanceObjectsAutoCompleterTest.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\Tests\Unit\Bundle\Maker\Factory;
+
+use PHPUnit\Framework\TestCase;
+use Zenstruck\Foundry\Bundle\Maker\Factory\NoPersistanceObjectsAutoCompleter;
+
+final class NoPersistanceObjectsAutoCompleterTest extends TestCase
```
Comment (nikophil) [inline tests/Unit/Bundle/Maker/Factory/NoPersistanceObjectsAutoCompleterTest.php:10]: I don't think (but I may be wrong) that it is possible to test this autocompletion behavior, even with `CommandCompletionTester`, this is why I used a unit test

**Comment by kbond** [comment]
Neat! Is the autocompletion "fuzzy" (can it match any part of the class name or just "starts with")?

**Comment by nikophil** [comment]
I would like to do this, but I couldn't: the autocompleter does not allow it. 

you can even pass a callback, but it does not take any argument
see https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Console/Question/Question.php#L134

**Comment by nikophil** [comment]
Hi @kbond 

any chance to see this validated/merged at some point? I don't know where I can simplify the code here :disappointed:

**Review by kbond — APPROVED** [review]

**Pull request merged**